### PR TITLE
[#3] Unit test configuration using catch2

### PR DESCRIPTION
**Pull request by Quik95**
Using packaged `catch2` from a package manager, because manually setting it up using make is too much of a pain. If we ever decide to migrate to cmake, we can consider migrating it to building `catch2` from source.

### CODE REVIEW

**Comment by jagtym** [comment]
> Using packaged `catch2` from a package manager, because manually setting it up using make is too much of a pain. 

Can you post here package version of catch2 package? I'll try to match it in flake.nix so we have exact same running in CI. 

> If we ever decide to migrate to cmake, we can consider migrating it to building `catch2` from source.

It there any reason for compiling it from source at the moment? Tests will mostly run in CI with pin-pointed version of the library.

Code looks good for me, approving!

**Review by jagtym — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -56,3 +59,15 @@ proto/proto.pb.o:
 .PHONY: proto-clean
 proto-clean:
 	rm -rf proto/proto.pb.o proto/*.pb.*
+
+.PHONY: test
```
Comment (mati2251) [inline Makefile:63]: For consistency, the test target should be named test-run and the test-binary target should be named test.

**Review by mati2251 — CHANGES_REQUESTED** [review]

**Review by mati2251 — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (mati2251) [reply Makefile:63]: @Quik95 .PHONY also needs to be changed. And `test-binary` should be `test`. I think that changes don't need new commit if before commits provide that stuff - similar too documentation commit. Personalny I would squash them to one new commit.

**Review by mati2251 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (mati2251) [reply Makefile]: I improved this stuff because these features are crucial to the project and need to be merged quickly.

**Review by mati2251 — APPROVED** [review]

**Pull request merged**